### PR TITLE
Deadline: better logging for DL webservice failures

### PR DESCRIPTION
**Pull request by kalisp**
## Brief description
It seems that DL webservice can return broken json payload even if `response.ok`. Safer and saner report.

## Additional info
Without it error message is a bit vague:
Screenshot 2022-08-18 171733

Not sure how to test it properly. I would just try to send regular DL submission from AE (or Harmony).

### CODE REVIEW

File: openpype/modules/deadline/abstract_submit_deadline.py [hunk 1]
```diff
@@ -627,7 +628,14 @@ def submit(self, payload):
             self.log.debug(payload)
             raise RuntimeError(response.text)
 
-        result = response.json()
+        try:
+            result = response.json()
+        except json.decoder.JSONDecodeError:
+            msg = "Broken response {}. ".format(response)
+            msg += "Try restarting DL webservice"
+            self.log.warning()
```
Comment (BigRoy) [inline openpype/modules/deadline/abstract_submit_deadline.py:636]: This seems to be missing the message.
```suggestion
            self.log.warning(msg)
```

File: openpype/modules/deadline/abstract_submit_deadline.py [hunk 2]
```diff
@@ -627,7 +628,14 @@ def submit(self, payload):
             self.log.debug(payload)
             raise RuntimeError(response.text)
 
-        result = response.json()
+        try:
+            result = response.json()
+        except json.decoder.JSONDecodeError:
+            msg = "Broken response {}. ".format(response)
+            msg += "Try restarting DL webservice"
```
Comment (BigRoy) [inline openpype/modules/deadline/abstract_submit_deadline.py:635]: I think abbreviating to Deadline to DL is just adding confusion in the clarity of the message.

```suggestion
            msg += "Try restarting the Deadline Webservice."
```

**Comment by BigRoy** [comment]
It's a tricky one. It's hard to tell what the culprit of the issue originally was and whether this PR will actually avoid the same confusion.

The fact that this now also raises its own replacement error I wonder whether it might make it harder to debug in the future than just re-raising the error with `raise` or alike. (Or at least having something that will follow that traceback?)

File: openpype/modules/deadline/abstract_submit_deadline.py [hunk 3]
```diff
@@ -627,7 +628,14 @@ def submit(self, payload):
             self.log.debug(payload)
             raise RuntimeError(response.text)
 
-        result = response.json()
+        try:
+            result = response.json()
+        except json.decoder.JSONDecodeError:
```
Comment (iLLiCiTiT) [inline openpype/modules/deadline/abstract_submit_deadline.py:633]: Exception `JSONDecodeError` in `json.decoder` is not available in Python 2 DCCs. Put this to the header and use `JSONDecodeError` here.
```
JSONDecodeError = getattr(json.decoder, "JSONDecodeError", ValueError)
```

File: openpype/modules/deadline/abstract_submit_deadline.py [hunk 4]
```diff
@@ -627,7 +628,14 @@ def submit(self, payload):
...
+            msg = "Broken response {}. ".format(response)
+            msg += "Try restarting DL webservice"
+            self.log.warning()
+            raise RuntimeError("Broken response from DL")
```
Comment (iLLiCiTiT) [inline openpype/modules/deadline/abstract_submit_deadline.py:637]: There is `KnownPublishError` in `openpype.pipeline.publish` for these cases.

File: openpype/modules/deadline/abstract_submit_deadline.py (hunk 1 above)
Comment (iLLiCiTiT) [reply openpype/modules/deadline/abstract_submit_deadline.py:636]: I did small edit in the suggestion to log traceback too.

File: openpype/modules/deadline/abstract_submit_deadline.py [hunk 5]
```diff
@@ -625,9 +631,16 @@ def submit(self, payload):
             self.log.error(response.status_code)
             self.log.error(response.content)
             self.log.debug(payload)
-            raise RuntimeError(response.text)
+            raise KnownPublishError(response.text)
+
+        try:
+            result = response.json()
+        except json.decoder.JSONDecodeError:
```
Comment (iLLiCiTiT) [inline openpype/modules/deadline/abstract_submit_deadline.py:638]:
```suggestion
        except JSONDecodeError:
```

**Review by iLLiCiTiT — APPROVED** [review]

**Pull request merged**